bulla-contracts/src/mappings/BullaFactoring.ts

```diff
@@ -258,3 +262,38 @@ export function handleSharesRedeemedWithAttachment(event: SharesRedeemedWithAtta
 
   sharesRedeemedEvent.save();
 }
+
+export function handleInvoiceImpaired(event: InvoiceImpaired): void {
+  const ev = event.params;
+  const originatingClaimId = ev.invoiceId;
+
+  const underlyingClaim = getClaim(originatingClaimId.toString());
+
+  const InvoiceImpairedEvent = createInvoiceImpairedEvent(originatingClaimId, event);
+
+  InvoiceImpairedEvent.invoiceId = underlyingClaim.id;
+  const price_per_share = getOrCreatePricePerShare(event);
+  const latestPrice = getLatestPrice(event);
+  const historical_factoring_statistics = getOrCreateHistoricalFactoringStatistics(event);
+
+  InvoiceImpairedEvent.eventName = "InvoiceImpaired";
+  InvoiceImpairedEvent.blockNumber = event.block.number;
+  InvoiceImpairedEvent.transactionHash = event.transaction.hash;
+  InvoiceImpairedEvent.logIndex = event.logIndex;
+  InvoiceImpairedEvent.fundedAmount = ev.lossAmount;
+  InvoiceImpairedEvent.impairAmount = ev.gainAmount;
+  InvoiceImpairedEvent.timestamp = event.block.timestamp;
+  InvoiceImpairedEvent.poolAddress = event.address;
+  InvoiceImpairedEvent.priceAfterTransaction = latestPrice;
+  InvoiceImpairedEvent.claim = underlyingClaim.id;
+  const lossAccrued = ev.lossAmount
+    .minus(underlyingClaim.paidAmount)
+    .minus(ev.gainAmount)
+    .neg();
+  const pool_pnl = getOrCreatePoolProfitAndLoss(event, lossAccrued);
+
+  pool_pnl.save();
```

Comment:
pool_pnl is also getting saved in "getOrCreatePoolProfitAndLoss"?

Reply:
good catch!